chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -2211,6 +2212,26 @@ impl PeerManagerActor {
     }
 }
 
+impl Handler<ActixMessageWrapper<PeerManagerMessageRequest, PeerManagerMessageResponse>>
+    for PeerManagerActor
+{
+    type Result = ActixMessageResponse<PeerManagerMessageResponse>;
+
+    fn handle(
+        &mut self,
+        msg: ActixMessageWrapper<PeerManagerMessageRequest, PeerManagerMessageResponse>,
+        ctx: &mut Self::Context,
+    ) -> Self::Result {
+        // Unpack throttle controller
+        let (msg, throttle_token) = msg.take();
+
+        let result = self.handle(msg, ctx);
+
+        // TODO(#5155) Add support for DeepSizeOf to result
+        ActixMessageResponse::new(result, ThrottleToken::new(throttle_token.into_inner(), 0))
```

Comment:
We can write this logic once in a generic way: 

```diff
diff --git a/chain/network/src/common/message_wrapper.rs b/chain/network/src/common/message_wrapper.rs
index ad66e948c..b2c028da1 100644
--- a/chain/network/src/common/message_wrapper.rs
+++ b/chain/network/src/common/message_wrapper.rs
@@ -1,5 +1,5 @@
 use actix::dev::MessageResponse;
-use actix::Message;
+use actix::{Actor, Handler, Message};
 use near_rate_limiter::{ThrottleController, ThrottleToken};
 use std::marker::PhantomData;
 
@@ -12,15 +12,14 @@ use std::marker::PhantomData;
 /// - Add needed decorators. Probably `Debug`, `Message` from Actix, etc.
 /// - Add two rate limiters (local per peer, global one)
 /// - Any other metadata we need debugging, etc.
-pub struct ActixMessageWrapper<T, Q> {
+pub struct ActixMessageWrapper<T> {
     msg: T,
     throttle_token: ThrottleToken,
-    ph: PhantomData<Q>,
 }
 
-impl<T, Q> ActixMessageWrapper<T, Q> {
+impl<T> ActixMessageWrapper<T> {
     pub fn new_without_size(msg: T, throttle_controller: ThrottleController) -> Self {
-        Self { msg, throttle_token: ThrottleToken::new(throttle_controller, 0), ph: PhantomData }
+        Self { msg, throttle_token: ThrottleToken::new(throttle_controller, 0) }
     }
 
     #[allow(unused)]
@@ -34,11 +33,20 @@ impl<T, Q> ActixMessageWrapper<T, Q> {
     }
 }
 
-impl<T, Q> Message for ActixMessageWrapper<T, Q>
-where
-    Q: 'static,
-{
-    type Result = ActixMessageResponse<Q>;
+impl<T: Message> Message for ActixMessageWrapper<T> {
+    type Result = ActixMessageResponse<T::Result>;
+}
+
+impl<T: Message> ActixMessageWrapper<T> {
+    pub fn handle<A: Actor + Handler<T, Result = T::Result>>(
+        self,
+        actor: &mut A,
+        ctx: &mut A::Context,
+    ) -> ActixMessageResponse<T::Result> {
+        let (msg, throttle_token) = self.take();
+        let result = actor.handle(msg, ctx);
+        ActixMessageResponse::new(result, ThrottleToken::new(throttle_token.into_inner(), 0))
+    }
 }
 
 #[derive(MessageResponse)]
diff --git a/chain/network/src/peer_manager/peer_manager_actor.rs b/chain/network/src/peer_manager/peer_manager_actor.rs
index 4cd7f2ec9..0808a6246 100644
--- a/chain/network/src/peer_manager/peer_manager_actor.rs
+++ b/chain/network/src/peer_manager/peer_manager_actor.rs
@@ -2212,23 +2212,15 @@ impl PeerManagerActor {
     }
 }
 
-impl Handler<ActixMessageWrapper<PeerManagerMessageRequest, PeerManagerMessageResponse>>
-    for PeerManagerActor
-{
+impl Handler<ActixMessageWrapper<PeerManagerMessageRequest>> for PeerManagerActor {
     type Result = ActixMessageResponse<PeerManagerMessageResponse>;
 
     fn handle(
         &mut self,
-        msg: ActixMessageWrapper<PeerManagerMessageRequest, PeerManagerMessageResponse>,
+        msg: ActixMessageWrapper<PeerManagerMessageRequest>,
         ctx: &mut Self::Context,
     ) -> Self::Result {
-        // Unpack throttle controller
-        let (msg, throttle_token) = msg.take();
-
-        let result = self.handle(msg, ctx);
-
-        // TODO(#5155) Add support for DeepSizeOf to result
-        ActixMessageResponse::new(result, ThrottleToken::new(throttle_token.into_inner(), 0))
+        msg.handle(self, ctx)
     }
 }
 
```

Not sure if we should hough -- in general, I personally don't like such type-heavy code.

Reply:
@matklad It's good suggestion. The answer is as always it depends. If it's only one use, then I wouldn't bother.
In the next PR, I'll include need more wrappers. We can decide what do it at that point.

Reply:
Yes, this works. However, I'll in the next PR I'll have to pass `throttle_controller` from ThrottleToken inside `actor.handle()`.  This generic will not work, I'll have to use something different.

```
-> ActixMessageResponse<T::Result> {
+        let (msg, throttle_token) = self.take();
+        let result = actor.handle(msg, ctx);
+        ActixMessageResponse::new(result, ThrottleToken::new(throttle_token.into_inner(), 0))
+    }
```